src/output.c

```diff
@@ -43,6 +43,19 @@ output_destroy_notify(struct wl_listener *listener, void *data)
 	wl_list_remove(&output->frame.link);
 	wl_list_remove(&output->destroy.link);
 
+	struct lab_layer_surface *layer, *tmp_layer;
+	int nr_layers = sizeof(output->layers) / sizeof(output->layers[0]);
+	for (int i = 0;i < nr_layers;i++) {
+		wl_list_for_each_safe (layer, tmp_layer, &output->layers[i], link) {
+			wlr_layer_surface_v1_destroy(layer->scene_layer_surface->layer_surface);
+		}
+	}
```

Comment:
Thanks for your PR.

I am not sure we need to destroy the layer surfaces here, they might be destroyed when their scene node gets destroyed. Need to verify this though. At the very least we should also remove the items from the list.

Reply:
They are destroyed in layers.c when the corresponding output is being destroyed.